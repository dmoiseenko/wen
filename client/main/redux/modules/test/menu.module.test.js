import deepFreeze from 'deep-freeze';

import * as module from '../menu.module';


it('reduce openMenu correctly', () => {
  const stateBefore = {
    menu: {
      isOpen: false
    }
  };

  deepFreeze(stateBefore);

  const actual = module.reducer[module.openMenu](stateBefore);

  expect(actual).toMatchSnapshot();
});

it('reduce closeMenu correctly', () => {
  const stateBefore = {
    menu: {
      isOpen: true
    }
  };

  deepFreeze(stateBefore);

  const actual = module.reducer[module.closeMenu](stateBefore);

  expect(actual).toMatchSnapshot();
});

it('reduce openCloseMenu correctly', () => {
  const stateBefore = {
    menu: {
      isOpen: false
    }
  };

  deepFreeze(stateBefore);

  const actual = module.reducer[module.openCloseMenu](stateBefore);

  expect(actual).toMatchSnapshot();
});
