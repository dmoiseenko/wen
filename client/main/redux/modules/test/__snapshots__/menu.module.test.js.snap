// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reduce closeMenu correctly 1`] = `
Object {
  "menu": Object {
    "isOpen": false,
  },
}
`;

exports[`reduce openCloseMenu correctly 1`] = `
Object {
  "menu": Object {
    "isOpen": true,
  },
}
`;

exports[`reduce openMenu correctly 1`] = `
Object {
  "menu": Object {
    "isOpen": true,
  },
}
`;
